client/site-profiler/components/domain-information/index.tsx

```diff
@@ -46,53 +48,72 @@ export default function DomainInformation( props: Props ) {
 		} );
 	};
 
+	const fWhois: { [ name: string ]: boolean } = {};
+
+	// Check if there are redacted whois fields
+	for ( const key in whois ) {
+		let value = whois[ key as keyof WhoIs ] ?? '';
+
+		if ( Array.isArray( value ) ) {
+			value = value.length > 0 ? value[ 0 ] : '';
+		}
+
+		value = value?.toLowerCase().replace( /[ .]/g, '' );
+		const isRedacted = redactedWhois.includes( value );
+
+		if ( isRedacted ) {
+			++fieldsRedacted;
+		}
+
+		fWhois[ key ] = ! isRedacted;
+	}
+
```

Comment:
Can we move this block to a separate hook file?

For example: `useFilteredWhoisData`.
I would like to see this component as simple as it can be (only presentational logic).

Reply:
Nice idea, thanks. Moved.